tests/step_methods/test_state.py

```diff
@@ -130,6 +131,11 @@ def test_sampling_state():
     with pytest.raises(AssertionError, match="Encountered invalid state class"):
         b1.sampling_state = b1_state.state1
 
+    with pytest.raises(
+        AssertionError, match="The supplied state is incompatible with the current sampling state."
+    ):
+        b1.sampling_state = b5.sampling_state
```

Comment:
Nit, I still dislike magic property. A method call is much more obvious that some checks may be done, and you're not just doing a random assignment. It also gives you a hatch to disable them if you're confident it's correct and want to avoid costly checks, because you're allowed to have kwargs then

Reply:
And you can't refactor properties into methods with back-compat. We had stuff with `model.logp` and it was a pita for saving two parenthesis back in the day